wp-admin/js/widgets/media-gallery-widget.js

```diff
@@ -0,0 +1,254 @@
+			defaultSync = wp.media.model.Attachment.prototype.sync;
+			wp.media.model.Attachment.prototype.sync = function rejectedSync() {
+				return $.Deferred().rejectWith( this ).promise();
+			};
+			mediaFrame.on( 'close', function onClose() {
+				wp.media.model.Attachment.prototype.sync = defaultSync;
+			});
+
+			mediaFrame.$el.addClass( 'media-widget' );
+			mediaFrame.open();
+
+			// Clear the selected attachment when it is deleted in the media select frame.
+			if ( selection ) {
+				selection.on( 'destroy', function onDestroy( attachment ) {
+					if ( control.model.get( 'attachment_id' ) === attachment.get( 'id' ) ) {
+						control.model.set({
+							attachment_id: 0,
+							url: ''
+						});
+					}
+				});
+			}
+		},
+		/**
+		 * Open the media select frame to chose an item.
+		 *
+		 * @returns {void}
+		 */
+		selectMedia: function selectMedia() {
+			var control = this, selection, mediaFrame, defaultSync, mediaFrameProps;
+			if ( control.isSelected() && 0 !== control.model.get( 'selection' ) ) {
+				selection = new wp.media.model.Selection( [ control.selectedAttachment ] );
+			} else {
+				selection = null;
+			}
+
+			mediaFrameProps = control.mapModelToMediaFrameProps( control.model.toJSON() );
+			if ( mediaFrameProps.size ) {
+				control.displaySettings.set( 'size', mediaFrameProps.size );
+			}
+			mediaFrame = new GalleryDetailsMediaFrame({
+				frame: 'select',
+				text: control.l10n.add_to_widget,
+				selection: selection,
+				mimeType: control.mime_type,
+				selectedDisplaySettings: control.displaySettings,
+				showDisplaySettings: control.showDisplaySettings,
+				metadata: mediaFrameProps,
+				state: 'gallery'
+			});
+			wp.media.frame = mediaFrame; // See wp.media().
+
+			// Handle selection of a media item.
+			mediaFrame.on( 'update', function onUpdate( selections ) {
+				var state = mediaFrame.state(), selectedImages;
+
+				selectedImages = selections || state.get( 'selection' );
+
+				if ( ! selectedImages ) {
+					return;
+				}
+
+				// Update widget instance.
+				control.model.set( {
+					ids: _.pluck( selectedImages.models, 'id' ).join( ',' ),
+					attachments: JSON.stringify(
+						selectedImages.models.map( function( model ) {
+							return model.toJSON();
+						} )
+					),
+					selection: selectedImages
+				} );
+			} );
+
+			// Disable syncing of attachment changes back to server. See <https://core.trac.wordpress.org/ticket/40403>.
+			defaultSync = wp.media.model.Attachment.prototype.sync;
+			wp.media.model.Attachment.prototype.sync = function rejectedSync() {
+				return $.Deferred().rejectWith( this ).promise();
+			};
+			mediaFrame.on( 'close', function onClose() {
+				wp.media.model.Attachment.prototype.sync = defaultSync;
+			});
+
+			mediaFrame.$el.addClass( 'media-widget' );
+			mediaFrame.open();
+
+			// Clear the selected attachment when it is deleted in the media select frame.
```

Comment:
Not sure we need this logic here... but will test to confirm

Reply:
It was needed. I confirmed.